trace/doc.go

```diff
@@ -0,0 +1,164 @@
+// To use the API, user code must set its service name via the global
+// variable Service, which is used by trace visualization tools to
+// tell apart the spans reported by different services. Code using the
+// trace API must set its Service, and set it only once, in main():
+//
+//    trace.Service = "my-awesome-service"
+//
+// The trace API is built on three abstractions: the
+// Client, the ClientBackend and the Span.
+//
+// Context
+//
+// Since it is very common for networked code to use context.Context,
+// veneur's trace API is designed to be used together with
+// Contexts. The API uses Contexts to store the current Span, and will
+// automatically pick the appropriate parent span from the Context.
+//
+// To effectively use the trace API, user code needs to ensure that
+// all functions that communicate externally (which includes using
+// traces) take Contexts as parameters, and pass them down.
+//
+// See https://blog.golang.org/context for an introduction to using
+// Context.
+//
+// Spans
+//
+// Spans represent the duration of a "unit of work" in a system: They
+// have a start and an end time, a name, and additional data that can
+// help identify what the work was. Example units of work could be the
+// handling of an HTTP request, or a database read. For convenience,
+// it is usually easiest to have a span represent a single function to
+// delineate the start and end, which makes it easy to create the Span
+// and report it via a defer statement.
+//
+// To create a new Span, use StartSpanFromContext. This function will
+// create a trace parented to the traces that came before it in the
+// Context. Typically, users should pass "" as the name to let
+// StartSpanFromContext figure out the correct name from caller
+// information.
+//
+//   span, ctx := trace.StartSpanFromContext(ctx, "")
+//   // report the span with the default client when this function returns:
+//   defer span.Finish()
+//   ... // do work here
+//
+// Reporting Spans via the trace Client
+//
+// Once constructed, SSF Spans must be reported to a system that
+// processes them. To do that, package trace exports a trace
+// Client. Typical applications will want to use the DefaultClient
+// exposed by this package. By default, it is set up to send spans to
+// veneur listening on the default SSF UDP port, 8128. An application
+// can use SetDefaultClient to change the default client in its main
+// function.
+//
+// To allow testing user code's Span reporting behavior, it is
+// desirable to take a Client argument in tested functions and report
+// spans to that client explicitly:
+//
+//   span, ctx := trace.StartSpanFromContext(ctx, "")
+//   defer span.ClientFinish(trace.DefaultClient)
+//
+// In case it is desired to submit no Spans at all, nil is a supported
+// trace Client value.
+//
+// Client Backends
+//
+// The trace Client can use various transports to send (and possibly
+// serialize) spans somewhere. Package trace exports some networked
```

Comment:
Reading this, I'm not sure what "send (and possibly serialize)" means - how does sending _not_ imply serialization?

Reply:
channel-backed backends like the one in the `testbackend` package in #582, for instance, never serialize, but ship the span off to another place!